File: backend/app/Models/CareProvider.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class CareProvider extends Model
{
    protected $fillable = [
        'user_id',
        'name',
        'specialty',
        'facility',
        'years_experience',
        'rating',
        'total_reviews',
        'bio',
        'languages',
    ];

    protected function casts(): array
    {
        return [
            'rating' => 'float',
            'years_experience' => 'integer',
            'total_reviews' => 'integer',
            'languages' => 'array',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Resolve the care_providers row for a directory user id, creating it on
     * first use. Admin screens identify doctors by their user id, while every
     * caseload gate (DoctorAccess) keys off care_providers.
     */
    public static function resolveForUser(int|string $userId): self
    {
        $user = User::find($userId);

        // `users.specialty` is nullable but `care_providers.specialty` is not,
        // so a doctor who never filled in a specialty used to blow this up with
        // a NOT NULL violation — surfacing as a 500 on the admin's "route this
        // request" action. Fall back to a neutral label instead.
        $specialty = trim((string) ($user?->specialty ?? ''));

        return static::updateOrCreate(
            ['user_id' => $userId],
            [
                'name' => $user?->fullName() ?? 'Care Provider',
                'specialty' => $specialty !== '' ? $specialty : 'General practice',
            ],
        );
    }

    public function requests(): HasMany
    {
        return $this->hasMany(CareRequest::class, 'provider_id');
    }

    public function assignments(): HasMany
    {
        return $this->hasMany(CareAssignment::class, 'provider_id');
    }

    public function toApiArray(): array
    {
        return [
            'id' => (string) $this->id,
            // The directory user behind the provider row. Every staff-facing
            // surface addresses clinicians by user id, so a payload that only
            // carries the care_providers id cannot name one — which is what
            // a patient asking a specific doctor for a document has to do.
            'user_id' => $this->user_id ? (string) $this->user_id : null,
            'name' => $this->name,
            'specialty' => $this->specialty,
            'facility' => $this->facility,
            'years_experience' => $this->years_experience,
            'rating' => $this->rating,
            'total_reviews' => $this->total_reviews,
            'bio' => $this->bio,
            'languages' => $this->languages ?? ['English'],
        ];
    }
}

File: frontend/lib/shared/models/care_provider.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';

enum CareRequestStatus { pending, approved, rejected, cancelled }

extension CareRequestStatusX on CareRequestStatus {
  String get label => switch (this) {
    CareRequestStatus.pending => 'Pending',
    CareRequestStatus.approved => 'Approved',
    CareRequestStatus.rejected => 'Rejected',
    CareRequestStatus.cancelled => 'Cancelled',
  };

  Color get color => switch (this) {
    CareRequestStatus.pending => AppColors.warning,
    CareRequestStatus.approved => AppColors.success,
    CareRequestStatus.rejected => AppColors.critical,
    CareRequestStatus.cancelled => AppColors.textMutedAA,
  };
}

class CareProvider {
  const CareProvider({
    required this.id,
    required this.name,
    required this.specialty,
    required this.facility,
    required this.yearsExperience,
    required this.rating,
    required this.totalReviews,
    this.userId,
    this.bio,
    this.languages = const ['English'],
    this.assigned = false,
  });

  final String id;

  /// The directory user behind this provider.
  ///
  /// Distinct from [id], which is the care_providers row. Every staff-facing
  /// endpoint addresses clinicians by user id, so this is what a patient
  /// naming a specific doctor — asking them for a document, say — has to send.
  /// Null for a provider row with no linked account.
  final String? userId;
  final String name;
  final String specialty;
  final String facility;
  final int yearsExperience;
  final double rating;
  final int totalReviews;
  final String? bio;
  final List<String> languages;
  final bool assigned;
}

class CareRequest {
  const CareRequest({
    required this.id,
    required this.providerId,
    required this.providerName,
    required this.providerSpecialty,
    required this.createdAt,
    required this.status,
    this.reason,
  });

  final String id;
  final String providerId;
  final String providerName;
  final String providerSpecialty;
  final DateTime createdAt;
  final CareRequestStatus status;
  final String? reason;
}
